clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -59,10 +60,14 @@ public synchronized void accept(R newValue, Throwable exception) {
                 future.completeExceptionally(exception);
             } else {
                 remainingResponses--;
-                if (remainingResponses <= 0)
-                    future.complete(null);
+                maybeComplete();
             }
         }
+
+        private void maybeComplete() {
+            if (remainingResponses <= 0)
```

Comment:
@cmccabe, is there a reason why we check for `<= 0` instead of `== 0`? It doesn't seem like negative numbers are ever possible. We could add an assert in the constructor to be sure. Generally, I prefer stricter checks so that bugs are caught instead of worked around. Thoughts?